llvm/lib/ObjectYAML/ELFEmitter.cpp

```diff
@@ -2090,6 +2099,9 @@ bool ELFState<ELFT>::writeELF(raw_ostream &OS, ELFYAML::Object &Doc,
   // Now we can decide segment offsets.
   State.setProgramHeaderLayout(PHeaders, SHeaders);
 
+  // Override section fields if requested.
```

Comment:
```suggestion
  // Override section fields, if requested. This needs to happen after program header layout happens, because otherwise the layout will use the new values.
```
(and reflow to column width rules of course)